<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>878</width>
    <height>397</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>QFDoubleEdit</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="3" column="4">
         <widget class="QSpinBox" name="spinIRangeMax">
          <property name="minimum">
           <number>-100000</number>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="value">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QSpinBox" name="spinIRangeMin">
          <property name="minimum">
           <number>-100000</number>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="value">
           <number>-100</number>
          </property>
         </widget>
        </item>
        <item row="3" column="5">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>increment:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>range:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>range:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QDoubleSpinBox" name="spinDRangeMax_2">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>-100000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100000.000000000000000</double>
          </property>
          <property name="value">
           <double>100.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QFDoubleEdit" name="edtInt"/>
        </item>
        <item row="2" column="3">
         <widget class="QDoubleSpinBox" name="spinDRangeMin_2">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>-100000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100000.000000000000000</double>
          </property>
          <property name="value">
           <double>0.001000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="6">
         <widget class="QDoubleSpinBox" name="spinDInc_2">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>-100000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100000.000000000000000</double>
          </property>
          <property name="value">
           <double>10.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="5">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>increment:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>double:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QFDoubleEdit" name="edtDouble"/>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>range:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QDoubleSpinBox" name="spinDRangeMin">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>-100000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100000.000000000000000</double>
          </property>
          <property name="value">
           <double>0.001000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QDoubleSpinBox" name="spinDRangeMax">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>-100000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100000.000000000000000</double>
          </property>
          <property name="value">
           <double>100.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="5">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>increment:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="6">
         <widget class="QDoubleSpinBox" name="spinDInc">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>-100000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100000.000000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>logdouble:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QFDoubleEdit" name="edtLogDouble"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>integer:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="6">
         <widget class="QSpinBox" name="spinIInc">
          <property name="minimum">
           <number>-100000</number>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="1">
         <widget class="QFDoubleSlider" name="horizontalSlider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>878</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QFDoubleEdit</class>
   <extends>QLineEdit</extends>
   <header>qfdoubleedit.h</header>
   <slots>
    <signal>valueChanged(double)</signal>
    <slot>setValue(double)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>QFDoubleSlider</class>
   <extends>QSlider</extends>
   <header>qfdoubleslider.h</header>
   <slots>
    <signal>dsliderMoved(double)</signal>
    <slot>setDValue(double)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>spinDRangeMin</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>541</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>592</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinDRangeMax</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>642</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>375</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinDInc</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>803</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>796</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinIRangeMin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>372</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinIRangeMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>703</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>386</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinIInc</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>855</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>dsliderMoved(double)</signal>
   <receiver>edtDouble</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>177</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>176</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edtDouble</sender>
   <signal>valueChanged(double)</signal>
   <receiver>horizontalSlider</receiver>
   <slot>setDValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateWidgets()</slot>
 </slots>
</ui>
